scripts/install_deps.py

```diff
@@ -0,0 +1,112 @@
+    )
+    stdout, stderr = p.communicate()
+    if p.returncode != 0:
+        output = (stdout + stderr).decode("utf-8")
+        raise BadRCError("Bad rc (%s) for cmd '%s': %s" % (p.returncode, cmd, output))
+    return stdout
+
+
+def _get_package_tarball(package_dir, package_prefix):
+    package_filenames = sorted(
+        [p for p in os.listdir(package_dir) if p.startswith(package_prefix)]
+    )
+    if len(package_filenames) == 0:
+        raise InstallationError(
+            "Unable to find local package starting with %s prefix."
+            % package_prefix
+        )
+    return package_filenames[-1]
+
+
+def pip_install_packages(package_dir):
+    package_dir = os.path.abspath(package_dir)
+
+    pip_script = os.path.join(os.environ["VIRTUAL_ENV"], "bin", "pip")
+    run("%s install --upgrade pip" % pip_script)
+
+    setup_requires_dir = os.path.join(package_dir, "setup")
+    with cd(setup_requires_dir):
+        _install_setup_deps(pip_script, ".")
+
+    cli_tarball = _get_package_tarball(package_dir, "awscli")
+    with cd(package_dir):
+        run(
+            "%s install %s --find-links file://%s %s"
+            % (pip_script, INSTALL_ARGS, package_dir, cli_tarball)
+        )
+
+
+def _install_setup_deps(pip_script, setup_package_dir):
+    # Some packages declare `setup_requires`, which is a list of dependencies
+    # to be used at setup time. These need to be installed before anything
+    # else, and pip doesn't manage them.  We have to manage this ourselves
+    # so for now we're explicitly installing the one setup_requires package
+    # we need.  This comes from python-dateutils.
+
+    # We need setuptools >= 37.0.0 for setuptools_scm to work properly
+    setuptools_tarball = _get_package_tarball(setup_package_dir, "setuptools-")
+    run(
+        "%s install --no-binary :all: --no-cache-dir --no-index "
+        "--find-links file://%s %s"
+        % (pip_script, setup_package_dir, setuptools_tarball)
+    )
+
+    setuptools_scm_tarball = _get_package_tarball(setup_package_dir, "setuptools_scm")
+    run(
+        "%s install --no-binary :all: --no-cache-dir --no-index "
+        "--find-links file://%s %s"
+        % (pip_script, setup_package_dir, setuptools_scm_tarball)
+    )
+    wheel_tarball = _get_package_tarball(setup_package_dir, "wheel")
+    run(
+        "%s install --no-binary :all: --no-cache-dir --no-index "
+        "--find-links file://%s %s" % (pip_script, setup_package_dir, wheel_tarball)
+    )
+
+
+def install_packages(package_dir):
+    if not _virtualenv_enabled():
+        raise InstallationError(
+            "Installation being performed outside of a virtualenv. Please enable before running."
+        )
+    else:
+        try:
+            pip_install_packages(package_dir)
+        except Exception as e:
+            raise InstallationError(
+                "Error encountered when installing CLI dependencies", e
+            )
```

Comment:
Can we use [sys.exc_info](https://docs.python.org/3.7/library/sys.html#sys.exc_info) here to not lose the full traceback context.

Reply:
Talked offline about it, but I'm fine with just not catching the exception here and let it propogate to help debugging. If we need to catch and run some logic on these later then we can go back and add this try except.